File: Java-SpringBoot/BE-02-DB/java-17-springboot-3.5/src/main/java/com/flyrank/taskservice/config/OpenAPIConfig.java
package com.flyrank.taskservice.config;

import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.info.Info;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class OpenAPIConfig {

    @Bean
    public OpenAPI customOpenAPI() {
        return new OpenAPI()
                .info(new Info()
                        .title("Flyrank BE-02 Task API (Java 17 & Spring Data JPA)")
                        .version("1.0.0")
                        .description("Task CRUD service backed by persistent H2/SQLite database with search, filtering, and stats."));
    }
}

File: Java-SpringBoot/BE-02-DB/kotlin-2.3-springboot-3.5/src/main/kotlin/com/flyrank/taskservice/config/OpenAPIConfig.kt
package com.flyrank.taskservice.config

import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Info
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class OpenAPIConfig {

    @Bean
    fun customOpenAPI(): OpenAPI {
        return OpenAPI()
            .info(
                Info()
                    .title("Flyrank BE-02 Task API (Kotlin 2.3 & Spring Data JPA)")
                    .version("1.0.0")
                    .description("Idiomatic Kotlin Task CRUD service with Spring Boot 3.4.2, Data Classes, and Spring Data JPA.")
            )
    }
}
